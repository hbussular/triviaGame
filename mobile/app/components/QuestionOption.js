import React, { Component } from 'react'
import { TouchableOpacity, Text } from 'react-native'
import PropTypes from 'prop-types'

import { colors } from '../styles'
import styles from './QuestionOption.style'

class QuestionOption extends Component {
  constructor(props) {
    super(props)

    this.state = {
      pressed: false,
    }
  }

  handlePress = () => {
    const {
      onPress,
      correct,
    } = this.props

    this.setState({
      pressed: true,
    })

    onPress({
      correctAwnser: correct,
    })
  }

  render() {
    const {
      answer,
      correct,
    } = this.props

    const {
      pressed,
    } = this.state

    return (
      <TouchableOpacity
        style={[
          styles.optionTouchContainer,
          { backgroundColor: pressed && correct ? colors.quarternary : colors.secondary },
        ]}
        onPress={this.handlePress}
      >
        <Text
          style={[
            styles.textContainer,
            { color: pressed && correct ? colors.secondary : colors.primary },
          ]}
        >
          {answer}
        </Text>
      </TouchableOpacity>
    )
  }
}

QuestionOption.defaultProps = {
  correct: false,
}

QuestionOption.propTypes = {
  answer: PropTypes.string.isRequired,
  correct: PropTypes.bool,
  onPress: PropTypes.func.isRequired,
}

export default QuestionOption
