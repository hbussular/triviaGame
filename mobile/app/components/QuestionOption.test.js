import React from 'react'
import renderer from 'react-test-renderer'
import QuestionOption from './QuestionOption'

test('renders correctly', () => {
  const props = {
    correct: true,
    answer: 'Mr. Potato Head',
    onPress: jest.fn(),
  }

  const tree = renderer
    .create(<QuestionOption {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
