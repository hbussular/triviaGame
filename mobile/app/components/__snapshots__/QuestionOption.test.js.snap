// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "borderRadius": 8,
      "marginBottom": 5,
      "opacity": 1,
      "paddingVertical": 20,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgba(213, 55, 43, 1)",
          "fontSize": 16,
          "fontWeight": "700",
          "textAlign": "center",
        },
        Object {
          "color": "rgba(213, 55, 43, 1)",
        },
      ]
    }
  >
    Mr. Potato Head
  </Text>
</View>
`;
